src/lib/parsers/sol-json.c

```diff
@@ -567,3 +567,144 @@ sol_json_is_valid_type(struct sol_json_scanner *scanner, enum sol_json_type type
            sol_json_scanner_skip_over(scanner, &token) &&
            token.end == last_position;
 }
+
+SOL_API int
+sol_json_token_get_unescaped_string(const struct sol_json_token *token, struct sol_buffer *buffer)
+{
+    int r;
+    const char *start, *p;
+    bool is_escaped = false;
+    char new_char;
+
+    SOL_NULL_CHECK(buffer, -EINVAL);
+    sol_buffer_init_flags(buffer, NULL, 0, SOL_BUFFER_FLAGS_NO_NUL_BYTE);
+
+    SOL_NULL_CHECK(token, -EINVAL);
+    SOL_NULL_CHECK(token->start, -EINVAL);
+    SOL_NULL_CHECK(token->end, -EINVAL);
+
+    if (sol_json_token_get_type(token) != SOL_JSON_TYPE_STRING) {
+        sol_buffer_init_flags(buffer, (char *)token->start,
+            sol_json_token_get_size(token), SOL_BUFFER_FLAGS_MEMORY_NOT_OWNED);
+        buffer->used = buffer->capacity;
+        return 0;
```

Comment:
This was implemented because conffile uses this to get the string value of other JSON types. I could create another function for that, or checking if the token is a JSON String, but I preferred this way.